### PR TITLE
WIP: Add how to for requesting generic permission

### PR DESCRIPTION
**Pull request by Ahmad-Elsayed**
Generic permission should allow the users to request any available device permission.

### CODE REVIEW

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+* Complete the [Prerequisites](/howto/mobile/deploying-native-app/#prerequisites) section of *How to Deploy Your First Mendix Native Mobile App*.
+* Install Mendix Studio Pro v9.10.0 and above to use the Native Mobile App Builder and the latest Nanoflow commons module.
+* Make sure your [Nanoflow Commons](/appstore/modules/nanoflow-commons/) module is up to date
+* Make sure you are using [Make It Native 9 app](releasenotes/mobile/make-it-native-parent/) and it is up-to-date.
+
+## 3 Using generic permission action
+
+Generic permission action can be used inside nanoflows in an existing app or make a new app from scratch. You can find it in the toolbox for a nanoflow in the Nanoflow commons section.
+
+### 3.1 permission name
```
Comment (jordanch) [inline content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: Nit: probably needs Sentence case

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+When using this action you need to choose the permission name you need. It should be one the available permissions in `NanoflowCommons.Enum_Permissions` enumeration.
+
+{{% alert color="info" %}}
+Keep in mind that the permission names are different for each platform, so for example if you need to request permission to access reading contacts, you will need to call the actions two times: one for Ios permission which is `CONTACTS_IOS` and another for the Android permission which is `READ_CONTACTS_ANDROID`.
```
Comment (jordanch) [inline content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: Nit: `iOS`

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+{{% /alert %}}
+
+### 3.2 Action return type
+
+This action returns the status of the permission after the action and it is of type `NanoflowCommons.Enum_PermissionStatus`. 
+These statuses are:
+|Permission status|Notes|
+|-----------------|-----|
+|`NanoflowCommons.Enum_PermissionStatus.granted`|The permission is granted after requesting the user or it was already granted previously|
```
Comment (jordanch) [inline content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: `...after requesting the user?` This sounds a bit strange. Can it be reworded? 
Maybe: after requesting permission from the user

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+When the permission is already blocked and you try to request it, a popup will appear to the user asking him to open the settings and grant the permission and you will get a `blocked` status.
+{{% /alert %}}
+
+{{% alert color="info" %}}
+These permissions statuses match the statuses that are received from the action, so you shouldn't override them.
+{{% /alert %}}
+
+### 3.3 Usage example
+
+This an example for requesting contact access permission for Android and IOS devices.
```
Comment (jordanch) [inline content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: Nit: `iOS`

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+- You can get the running platform (Android or IOS) through using `Get device info` action from Native mobile section in the Toolbox.
```
Comment (jordanch) [inline content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: Nit: `iOS`

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+- Then you can branch your nanoflow using a decision which check for the system name.
```
Comment (jordanch) [inline content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: I suggest rewording this to:
`then you can branch your logic based on the current platform`

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+- Then for Android branch, you can use Generic permission action with a permission name of `READ_CONTACTS_ANDROID`.
```
Comment (jordanch) [inline content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: I suggest `of e.g. <permission name>`

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+- For Ios branch, you can also use a generic permission action with a permission name of `CONTACTS_IOS`.
```
Comment (jordanch) [inline content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: I suggest `of .e.g. <name>`

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+	{{< figure src="/attachments/howto/mobile/native-mobile/implementation/use-generic-permission/fullExample" alt="use generic permission screenshoot"   width="400"  >}}
+	{{< figure src="/attachments/howto/mobile/native-mobile/implementation/use-generic-permission/decision" alt="use generic permission screenshoot"   width="400"  >}}
+	{{< figure src="/attachments/howto/mobile/native-mobile/implementation/use-generic-permission/android" alt="use generic permission screenshoot"   width="400"  >}}
+	{{< figure src="/attachments/howto/mobile/native-mobile/implementation/use-generic-permission/ios" alt="use generic permission screenshoot"   width="400"  >}}
+
+## Deployment
```
Comment (jordanch) [inline content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: Is this part describing deployment or native app building? IMO, it's more about building that deployment

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+During deployment you must specify the permissions that your application has used to be able to request them from the user on production.
+To do that you can navigate to App permissions while using native ui builder and add the permissions you are using in your application for Ios and Android.
```
Comment (jordanch) [inline content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: Nit: `Native Builder UI`

Comment (jordanch) [reply content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: Nit: `iOS`

**Review by jordanch — CHANGES_REQUESTED** [review]
Good job 👏

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md (hunk 1 above)
Comment (Ahmad-Elsayed) [reply content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: Done

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md (hunk 2 above)
Comment (Ahmad-Elsayed) [reply content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: done

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md (hunk 3 above)
Comment (Ahmad-Elsayed) [reply content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: done

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md (hunk 4 above)
Comment (Ahmad-Elsayed) [reply content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: done

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md (hunk 5 above)
Comment (Ahmad-Elsayed) [reply content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: done

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md (hunk 6 above)
Comment (Ahmad-Elsayed) [reply content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: done

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md (hunk 7 above)
Comment (Ahmad-Elsayed) [reply content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: done

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md (hunk 8 above)
Comment (Ahmad-Elsayed) [reply content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: done

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md (hunk 10 above)
Comment (Ahmad-Elsayed) [reply content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: done

File: content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md (hunk 9 above)
Comment (Ahmad-Elsayed) [reply content/en/docs/howto/mobile/native-mobile/implementation/use-generic-permission.md]: done

**Review by jordanch — APPROVED** [review]

File: content/en/docs/appstore/modules/nanoflow-commons.md [hunk 11]
```diff
@@ -42,6 +42,7 @@ The [Nanoflow Commons](https://marketplace.mendix.com/link/component/109515/) mo
 |                     | Get object by guid           |
 |                     | Get platform                 |
 |                     | Wait                         |
+| Permissions         | RequestGenericPermission     |
```
Comment (pratikgoyal9) [inline content/en/docs/appstore/modules/nanoflow-commons.md]: @Ahmad-Elsayed Should be part of "Native mobile"

**Review by pratikgoyal9 — CHANGES_REQUESTED** [review]
